Mage.Sets/src/mage/cards/a/AwakenTheSleeper.java

```diff
@@ -0,0 +1,91 @@
+package mage.cards.a;
+
+import mage.abilities.Ability;
+import mage.abilities.condition.common.TargetPermanentIsEquippedCondition;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.UntapTargetEffect;
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.abilities.effects.common.continuous.GainControlTargetEffect;
+import mage.abilities.keyword.HasteAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetCreaturePermanent;
+
+import java.util.LinkedList;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class AwakenTheSleeper extends CardImpl {
+    public AwakenTheSleeper(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{3}{R}");
+
+        //Gain control of target creature until end of turn. Untap that creature. It gains haste until end of turn.
+        //If it's equipped, you may destroy all Equipment attached to that creature.
+        this.getSpellAbility().addEffect(new GainControlTargetEffect(Duration.EndOfTurn));
+        this.getSpellAbility().addEffect(new UntapTargetEffect().setText("Untap that creature"));
+        this.getSpellAbility().addEffect(new GainAbilityTargetEffect(
+                HasteAbility.getInstance(), Duration.EndOfTurn
+        ).setText("It gains haste until end of turn."));
+        this.getSpellAbility().addEffect(new ConditionalOneShotEffect(
```

Comment:
this doesn't need to be a `ConditionalOneShotEffect`, you're already using a custom effect